virtio-devices/src/thread_helper.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright © 2021 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::{
+    seccomp_filters::{get_seccomp_filter, Thread},
+    ActivateError,
+};
+use seccompiler::{apply_filter, SeccompAction};
+use std::{
+    panic::AssertUnwindSafe,
+    thread::{self, JoinHandle},
+};
+use vmm_sys_util::eventfd::EventFd;
+
+pub(crate) fn spawn_virtio_thread<F>(
+    name: &str,
+    seccomp_action: &SeccompAction,
+    thread_type: Thread,
+    epoll_threads: &mut Vec<JoinHandle<()>>,
+    exit_evt: &EventFd,
+    f: F,
+) -> Result<(), ActivateError>
+where
+    F: FnOnce(),
+    F: Send + 'static,
+{
+    let seccomp_filter = get_seccomp_filter(seccomp_action, thread_type)
+        .map_err(ActivateError::CreateSeccompFilter)?;
+
+    let panic_exit_evt = exit_evt.try_clone().unwrap();
```

Comment:
Would it be worth the time to add an error for the try_clone failure? We get an error when this is attempted in the device manager and this is called in a lot of places now.